### PR TITLE
Deploy gateway, organization, anchor and UT given a EIP20 token in origin chain

### PR DESCRIPTION
**Pull request by sarvesh-ost**
This PR is about setting up gateways and UBT given a token in origin chain. 

Index.js deploys the contracts and write address to a contracts.json file in the root directory

This implementation is in an early stage. Some naming and refactoring are definitely needed. But this PR will help unlock anyone working on the facilitator. Meanwhile, I can iterate over it. 

In index.js `numberOfDeployment` variable decided how many times deployment should be run. It appends address in `contracts.json`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 node_modules
```
Comment (pgev) [inline .gitignore]: More an alignment topic (unimportant one): we should have a template .gitignore to copy and not got annoyed :).

File: config/chain_config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+class ChainConfig {
```
Comment (pgev) [inline config/chain_config.js]: A proposal to evaluate: to make it readable and also to separate concepts, we can have to separate config representations for origin and auxiliary. Something like this:

config.origin.ipcProvider
config.origin.chainID
config.origin.contractAddresses.gateway

config.auxiliary.contractAddresses.coGateway

Comment (pgev) [inline config/chain_config.js]: Another proposal, we can have a configuration file with default options saved in repo and use it to fill config object.

File: src/deployer.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      txOptions: {
+        gasPrice: ChainConfig.auxiliaryGasPrice,
+      },
+      burner: ChainConfig.auxiliaryBurnerAddress,
+      masterKey: ChainConfig.auxiliaryMasterKey,
+    };
+  }
+
+  _deployOrganization() {
+    return Setup.organizations(
```
Comment (benjaminbollen) [inline src/deployer.js]: note for mosaic.js: the organization setup should probably just be done for a singular organization on a given chain.

Comment (schemar) [reply src/deployer.js]: For that there is `Organization.deploy()`

**Pull request merged**